contracts/apps/disputable/IAgreement.sol

```diff
@@ -5,11 +5,12 @@
 pragma solidity ^0.4.24;
 
 import "../../acl/IACLOracle.sol";
-import "../../lib/token/ERC20.sol";
 import "../../lib/arbitration/IArbitrable.sol";
 import "../../lib/arbitration/IAragonAppFeesCashier.sol";
+import "../../lib/token/ERC20.sol";
 
 
+// Note
```

Comment:
This is my primary note.

I've started thinking about the `IAgreement` as a "disputable action registry" (`IDisputableActionRegistry`), and backtracking on whether we need everything in this interface.

From a Disputable's standpoint, I think the main things its "action registry" should care about are:
- Registering new actions, which may be disputable
- Challenging an action
- Closing old actions
- Getting a little bit of information about a particular action or challenge

With that in mind, I started thinking that specific implementation-specific information about the "action registry" would be reduced and queried separately from the action disputable's information (e.g. when using subgraphs). For example, I noticed you only used the Agreement to find the vote's ID when disputes were created in the DisputableVoting subgraph.

Reply:
We decided not to force the `IArbitrable` and `IACL` inheritance at this level. The rest of the changes will be included in the aragonOS v6 roadmap